src/Network/WebSockets/Client.hs

```diff
@@ -92,14 +94,35 @@ runClientWithStream
     -- ^ Client application
     -> IO a
 runClientWithStream stream host path opts customHeaders app = do
+    newClientConnection stream host path opts customHeaders >>= app
+
+-- | Build a new 'Connection' from the client's point of view.
+newClientConnection
+    :: Stream
+    -- ^ Stream that will be used by the new 'Connection'.
+    --
+    -- /WARNING/: Be sure to call 'Stream.close' on this stream after you are
+    -- done using the 'Connection' in order to properly close the communication
+    -- channel. 'runClientWithStream' handles this for you, prefer to use it
+    -- when possible.
```

Comment:
Can you move this warning to the top-level comment rather than the comment attached to the argument?  That way it's a little more visible.